osfclient/models/storage.py

```diff
@@ -76,13 +77,18 @@ def create_file(self, path, fp, update=False):
         # peek at the file to check if it is an ampty file which needs special
         # handling in requests. If we pass a file like object to data that
         # turns out to be of length zero then no file is created on the OSF
+        conn_error = False
         if file_empty(fp):
             response = self._put(url, params={'name': fname}, data=b'')
         else:
-            response = self._put(url, params={'name': fname}, data=fp)
-
-        if response.status_code == 409:
+            try:
```

Comment:
I'd add a comment here along the lines of: "When uploading a large file (>a few MB) that already exists we sometimes get a `ConnectionError` instead of a status == 409." Mostly so we can remember in the future why we did this.

Reply:
Also add a link to this PR in the code?

Reply:
Sounds like a good idea.